### ISSUE TITLE
theme: Add different ticks length for different axes

### ISSUE THREAD

**Issue by pank** [tidyverse/ggplot2#2935]
Hi,

I would like to add support for different ticks length for different axes (basically extending `axis.ticks.length`).
My main motivation is to match an Excel theme that I have been given, which features upwards ticks on the x-axis. I don't want inwards ticks on the y-axis though.

It was first requested in issue #1319.

Related pull request [2934](https://github.com/tidyverse/ggplot2/pull/2934).

My answers to comments submitted to the pull requests:

@batpigandme wrote:
>  PS As for styling, you can use the styler package to implement the in-house style as described in the tidyverse style guide.

Thanks, I will check that.  I think it might also be possible to configure ESS to do tidyverse style these days...

@clauswilke 
> I've wanted this option in the past, so I'd generally be in favor. Not sure if it can be done without the clutter of adding all these new theme elements.

That's a good idea.  We could use a "list" like for `margins`.  Would it make sense to allow `axis.ticks.length` to be *either* a `unit` *or* a "`list`" with `top`, `bottom`, `left`, `right`?  Is there any precedence for allowing two types?  
The good things about having a new element in `theme` for each is that it makes auto-completion easy (at least in ESS and I'd assume Rstudio as well, but I'm not sure).

> For the theme calculation code, you could probably just write: ...

Thanks!  I could see if I `within` or `with` works here.  That would make is less cluttered as well...

**Comment by clauswilke** [comment]
There's also another issue, more a question for @hadley: We have a solid inheritance system for elements, but we don't for units, such as `axis.ticks.length` discussed here or similarly `legend.spacing` and probably a few others. We always handle unit inheritance in a one-off fashion, as in:
``` r
theme$axis.ticks.length.x.bottom <- theme$axis.ticks.length.x.bottom %||% theme$axis.ticks.length.x %||% theme$axis.ticks.length
```
Would it make sense to add a function that can calculate these units based on the theme and the inheritance tree?

**Comment by pank** [comment]
>Would it make sense to add a function that can calculate these units
>based on the theme and the inheritance tree?

I'm happy to add something like this if you point me in the right direction.

(I must admit that I was a bit surprised that inheritance did not occur "automatically" given the second argument to `el_def`).

**Comment by clauswilke** [comment]
Closed via #2934.
